_posts/2019-09-10-businesses-dont-fail-they-make-mistakes.md

```diff
@@ -12,51 +12,51 @@ tags:
 
-Being a message based system the message retry engine was kicking in and eventually failed messages were ending up in the error queue.
+Being a message-based system, the message retry engine was kicking in and eventually failed messages were ending up in the error queue.
 
 The root cause was clear and pretty straightforward. However I asked the same question again:
 
-> How come that a business failure caused a message to end up in the error queue?
+> How come a business failure caused a message to end up in the error queue?
 
 At which point, I had their attention.
 
 ## Business failures are not errors
 
-When dealing with failed messages in a message based system there are two important questions we should ask:
+When dealing with failed messages in a message-based system there are two important questions we should ask:
 
-* Is there any good reason to retry, in a tight loop or manually later, a message failing due to a well known business failure?
-* Is there something that an Operations Team can do, other that stating the obvious, when faced with a message that failed for a business reason?
+* Is there any good reason to retry, in a tight loop or manually later, a message failing due to a well-known business failure?
+* Is there something that an Operations Team can do, other than stating the obvious, when faced with a message that failed for a business reason?
 
-In a scenario, such as the above one, there is no point is retrying the create `cash order` request, given its actual state, it will always fail. Blindly retrying the failed message will cause an infinite loop, it'll fail again, and again, and again.
+In a scenario, such as the above one, there is no point in retrying the create `cash order` request, given its actual state; it will always fail. Blindly retrying the failed message will cause an infinite loop, it'll fail again, and again, and again.
 
 ### Design for (business) failures
 
-Designing for failures is on of the leading mantra when it comes to distributed systems. The same approach should be applied to business scenarios by modelling failures using messages.
+Designing for failures is one of the leading mantras when it comes to distributed systems. The same approach should be applied to business scenarios by modelling failures using messages.
 
-In the mentioned case what the back-end services should do is reply to the request originator with a specific message describing the failure encountered and, if required, compensating actions that can be performed to fix the business rule violation.
+In the mentioned case, what the back-end services should do is reply to the request originator with a specific message describing the failure encountered and, if required, compensating actions that can be performed to fix the business rule violation.
```

Comment:
Consider putting in an example. It seems like there is a lot of setup to explain the problem and then the solution is wrapped up very quickly.